Libraries/Microsoft.Teams.Plugins/Microsoft.Teams.Plugins.AspNetCore.BotBuilder/Controllers/MessageController.cs

```diff
@@ -53,10 +50,9 @@ public async Task<IResult> PostAsync()
                 return Results.Empty;
             }
 
-            // Fallback logic
-            var authHeader = HttpContext.Request.Headers.Authorization.FirstOrDefault() ?? throw new UnauthorizedAccessException();
-            var token = new JsonWebToken(authHeader.Replace("Bearer ", ""));
-            var res = await _plugin.Do(new()
+            // Fallback logic use the plugin to process the activity
+            var token = _plugin.ExtractToken(HttpContext.Request);
+            var res = await _plugin.Do(new ActivityEvent()
```

Comment:
since we now have the Do which takes in the request should we use that here to minimize code?